lezioni/riepargomgendata.php

```diff
@@ -232,17 +247,48 @@
                             <td width=35%>Attivit&agrave;</td>";
 
                     while ($reclez = mysqli_fetch_array($rislez)) {
-                        print "<tr><td>" . $docassoclist[$reclez['iddocente']] . "</td><td>" . $matassoclist[$reclez['idmateria']] . "</td><td>" . $reclez['argomenti'] . "&nbsp;</td><td>" . $reclez['attivita'] . "&nbsp;</td></tr>";
+                        stampa_lez(
+                            $con,
+                            $docassoclist,
+                            $reclez['idlezione'], 
+                            $matassoclist[$reclez['idmateria']],
+                            $reclez['argomenti'],
+                            $reclez['attivita']
+                        );
                     }
 
                     print "</table>";
                 }
             }
         }
-        // VISUALIZZARE ARGOMENTI SOSTEGNO
     }
 }
 
 mysqli_close($con);
 
 stampa_piede("");
+
+function stampa_lez($con, $docassoclist, $idlezione, $materia, $argomenti, $attvità)
+{
+    print "<tr>";
+    //selezioniamo tutti i docenti che hanno firmato quella lezione
+    $lezquery = "SELECT `iddocente` FROM `tbl_firme` WHERE `idlezione` = $idlezione;";
+    $risdocs = eseguiQuery($con, $lezquery);
+    if(mysqli_num_rows($risdocs) == 0){
+        print "<td>Nessuna firma</td>";
+    }else{
+        //stampa tutti i docenti corrispondenti alla lezione
+        print "<td>";
+        $f = true;
+        while($docente = mysqli_fetch_array($risdocs)){
+            //stampa la virgola solo se ci sono più docenti
+            if(!$f)
+                print ", ";
+            $f = false;
+            print $docassoclist[$docente['iddocente']];
+        }
+        print "</td>";
+    }
+    //stampa il resto della tabella
+    print "<td>" . $materia . "</td><td>" . $argomenti . "&nbsp;</td><td>" . $attvità . "&nbsp;</td></tr>";
```

Comment:
Sperando di dare un contributo utile, come accennato nell'altra PR, per evitare XSS credo sia necessario questo in questa riga di LAMPSchool:

```
$materia   = htmlentities( $materia   );
$argomenti = htmlentities( $argomenti );
$attvità   = htmlentities( $attvità   );
print "<td>" . $materia . "</td><td>" . $argomenti . "&nbsp;</td><td>" . $attvità . "&nbsp;</td></tr>";
```

Reply:
Naturalmente se i dati di provenienza sono già stati passati dentro `htmlentities()` allora si può ignorare il mio suggerimento ma meglio lasciare un commento in linea per semplificare futuri auditing.

Reply:
Giusto, uso la funzione interna a LAMPSchool fatta apposta per questo, ovvero `stringa_html()`